src/pages/Pools/Pair/Pair.tsx

```diff
@@ -0,0 +1,190 @@
+import { JSBI, Percent } from '@swapr/sdk'
+
+import React, { useCallback, useState } from 'react'
+import { ChevronDown } from 'react-feather'
+import { useTranslation } from 'react-i18next'
+import Skeleton from 'react-loading-skeleton'
+import { Link, Redirect, RouteComponentProps } from 'react-router-dom'
+import { Box, Flex, Text } from 'rebass'
+import styled from 'styled-components'
+
+import { ButtonBadge, ButtonPurpleDim } from '../../../components/Button'
+import DoubleCurrencyLogo from '../../../components/DoubleLogo'
+import List from '../../../components/LiquidityMiningCampaigns/List'
+import { DimBlurBgBox } from '../../../components/Pool/DimBlurBgBox/styleds'
+import { PoolStats } from '../../../components/Pool/PairView/PoolStats'
+import { UserLiquidity } from '../../../components/Pool/PairView/UserLiquidity'
+import { ValueWithLabel } from '../../../components/Pool/PairView/ValueWithLabel'
+import { RowBetween, RowFixed } from '../../../components/Row'
+import { PairSearchModal } from '../../../components/SearchModal/PairSearchModal'
+import { PairState, usePair } from '../../../data/Reserves'
+import { useToken } from '../../../hooks/Tokens'
+import { usePairLiquidityMiningCampaigns } from '../../../hooks/usePairLiquidityMiningCampaigns'
+import { useRouter } from '../../../hooks/useRouter'
+import { unwrappedToken } from '../../../utils/wrappedCurrency'
+import { PageWrapper } from '../../PageWrapper'
+
+export function Pair({
+  match: {
+    params: { currencyIdA, currencyIdB },
+  },
+}: RouteComponentProps<{ currencyIdA: string; currencyIdB: string }>) {
+  const router = useRouter()
+  const token0 = useToken(currencyIdA)
+  const token1 = useToken(currencyIdB)
+
+  const wrappedPair = usePair(token0 || undefined, token1 || undefined)
+  const [openPairsModal, setOpenPairsModal] = useState(false)
+  const { t } = useTranslation()
+  const { loading: loadingPairs, miningCampaigns } = usePairLiquidityMiningCampaigns(
+    wrappedPair[1] ? wrappedPair[1] : undefined
+  )
+
+  const handleAllClick = useCallback(() => {
+    setOpenPairsModal(true)
+  }, [])
+
+  const handleModalClose = useCallback(() => {
+    setOpenPairsModal(false)
+  }, [])
+
+  const handlePairSelect = useCallback(
+    pair => {
+      router.push({
+        pathname: `/pools/${pair.token0.address}/${pair.token1.address}`,
+      })
+    },
+    [router]
+  )
```

Comment:
just thinking, maybe we can pass only addresses instead of pair?
```suggestion
  const handlePairSelect = useCallback(
    (token0Address,token1Address) => {
      router.push({
        pathname: `/pools/${token0Address}/${token1Address}`,
      })
    },
    [router]
  )
```

Reply:
` <PairSearchModal isOpen={openPairsModal} onDismiss={handleModalClose} onPairSelect={handlePairSelect} />`

This is used on `PairSearchModal` and there it will pass a `Pair` to `onPairSelect` which is the function we pass.
So we can't pass the addresses here. 

<img width="397" alt="image" src="https://user-images.githubusercontent.com/5664434/174296285-73ad94d6-4898-4cfa-8121-9b15d92e5723.png">

However we can destruct it, which might be cleaner 

```
({token0, token1}) => {
      router.push({
        pathname: `/pools/${token0.address}/${token1.address}`,
      })
    },
```